File: apps/web/lib/reviews.ts
export type ReviewStatus = "pending" | "approved" | "rejected" | "changes_requested";
export type ReviewDecision = "approved" | "rejected" | "changes_requested";

export type DemoReview = {
  id: string;
  title: string;
  status: ReviewStatus;
  priority: "low" | "medium" | "high";
  sourceDocumentId: string;
  aiRunId: string;
  createdAt: string;
  reviewerGuidance: string;
};

export const demoReviews: DemoReview[] = [
  {
    id: "review_vendor_risk_summary",
    title: "Review AI vendor risk summary",
    status: "pending",
    priority: "high",
    sourceDocumentId: "doc_vendor_intake",
    aiRunId: "airun_vendor_risk_summary",
    createdAt: "2026-01-15T12:12:00Z",
    reviewerGuidance:
      "Approve only if the answer cites vendor risk source snippets and does not invent controls."
  },
  {
    id: "review_contract_exception",
    title: "Confirm contract exception classification",
    status: "pending",
    priority: "medium",
    sourceDocumentId: "doc_contract_notes",
    aiRunId: "airun_contract_exception",
    createdAt: "2026-01-15T12:25:00Z",
    reviewerGuidance:
      "Request changes if the extraction is incomplete or the classification lacks support."
  },
  {
    id: "review_policy_gap_completed",
    title: "Approved policy gap summary",
    status: "approved",
    priority: "low",
    sourceDocumentId: "doc_policy_update",
    aiRunId: "airun_policy_gap_check",
    createdAt: "2026-01-15T10:15:00Z",
    reviewerGuidance: "Completed review retained as audit evidence."
  }
];

export function getReviews(): DemoReview[] {
  return demoReviews;
}

export function getReviewById(reviewId: string): DemoReview | undefined {
  return demoReviews.find((review) => review.id === reviewId);
}

File: apps/api/app/services/reviews.py
from datetime import UTC, datetime

from app.models.reviews import ReviewDecisionRequest, ReviewDecisionResult, ReviewItem


def get_demo_reviews() -> list[ReviewItem]:
    return [
        ReviewItem(
            id="review_vendor_risk_summary",
            title="Review AI vendor risk summary",
            status="pending",
            priority="high",
            source_document_id="doc_vendor_intake",
            ai_run_id="airun_vendor_risk_summary",
            created_at=datetime(2026, 1, 15, 12, 12, tzinfo=UTC),
        ),
        ReviewItem(
            id="review_contract_exception",
            title="Confirm contract exception classification",
            status="pending",
            priority="medium",
            source_document_id="doc_contract_notes",
            ai_run_id="airun_contract_exception",
            created_at=datetime(2026, 1, 15, 12, 25, tzinfo=UTC),
        ),
        ReviewItem(
            id="review_policy_gap_completed",
            title="Approved policy gap summary",
            status="approved",
            priority="low",
            source_document_id="doc_policy_update",
            ai_run_id="airun_policy_gap_check",
            created_at=datetime(2026, 1, 15, 10, 15, tzinfo=UTC),
        ),
    ]


def get_demo_review(review_id: str) -> ReviewItem | None:
    return next((review for review in get_demo_reviews() if review.id == review_id), None)


def decide_review(
    review: ReviewItem,
    request: ReviewDecisionRequest,
) -> ReviewDecisionResult:
    if review.status != "pending":
        raise ValueError("Review decision cannot be changed after it leaves the pending state.")

    decided_review = review.model_copy(update={"status": request.decision})
    return ReviewDecisionResult(
        review=decided_review,
        reviewer_email=request.reviewer_email,
        reviewer_note=request.reviewer_note,
        decided_at=datetime.now(UTC),
        audit_event=f"review.{request.decision}",
    )
